packages/recomponents/src/components/r-pagination/r-pagination.vue

```diff
@@ -103,54 +93,56 @@
         },
         computed: {
             items() {
-                const {totalVisible} = this;
+                const items = {};
+                const setPageItem = (index) => {
+                    const page = {
+                        content: index + 1,
+                        selected: index === (this.page - 1),
+                    };
+                    items[index] = page;
+                };
 
-                if (!totalVisible) {
-                    return null;
-                }
+                if (this.pages <= this.totalVisible) {
+                    for (let index = 0; index < this.pages; index += 1) {
+                        setPageItem(index);
+                    }
+                } else {
+                    const half = Math.floor(this.totalVisible / 2);
 
-                if (this.pages <= totalVisible || totalVisible < 1) {
-                    return this.range(1, this.pages);
-                }
+                    const setBreakView = (index) => {
+                        const breakView = {breakView: true};
+                        items[index] = breakView;
+                    };
 
-                const even = totalVisible % 2 === 0 ? 1 : 0;
-                const left = Math.floor(totalVisible / 2);
-                const right = this.pages - left + 1 + even;
+                    let left = 0;
+                    if (this.page - half > 0) {
+                        left = this.page - 1 - half;
+                    }
 
-                if (this.page > left && this.page < right) {
-                    const start = this.page - left + 1;
-                    const end = this.page + left - 1 - even;
+                    let right = left + this.totalVisible - 1;
+                    if (right >= this.pages) {
+                        right = this.pages - 1;
+                        left = right - this.totalVisible + 1;
+                    }
 
-                    if (totalVisible === 1 || totalVisible === 2) {
-                        const items = [];
-                        if (totalVisible === 2) {
-                            items.push(1);
-                        }
-                        if (this.page !== 1 && (this.page !== 2 && totalVisible === 2)) {
-                            items.push('...');
+                    for (let i = left; i <= right && i <= this.pages - 1; i += 1) {
```

Comment:
I now that you just modified the existing` items()` computed method but I find it too complex. I would try to simplify it so it's easier to understand (and therefore improve in future iterations). 
Good news that we already have some tests  in place (although I'm not sure if they are complete enough) so we can improve the method with a bit more of confidence. 

For example:

Could we make this expression more expressive: `for (let i = left; i <= right && i <= this.pages - 1; i += 1)` ?
Could we replace it by a forEach or even map? I'm sure that we can. Is it the most urgent thing to do now? I don't know, the more changes/bugs a piece of code absorbs, the more useful will be to refactor it looking for simplicity/reusability.  

I think that it would be a good exercise to work a bit in this concrete method to understand the type of refactors that I'd suggest. Tell me if you want to try it now or maybe in the future.

Reply:
Created a new task to simplify the RPagination component.